trainer.py

```diff
@@ -100,7 +99,7 @@ def evaluate_model_on_dataloader(
         total_loss = 0
         avg_loss = 0
         accuracy = 0
-        nof_samples = 0
+        nof_samples = len(dataset)
```

Comment:
hange back to 0, and update during batch iteration
nof_samples += self.batch_size (or something like that)